core/app/services/spree/export.rb

```diff
@@ -0,0 +1,103 @@
+require 'nokogiri'
+
+module Spree
+  class Export
+    prepend Spree::ServiceModule::Base
+
+    def export_google_rss(options)
+      store = Spree::Store.find(options.store)
+      @options = options
+
+      builder = Nokogiri::XML::Builder.new do |xml|
+        xml.rss('xmlns:g' => 'http://base.google.com/ns/1.0', 'version' => '2.0') do
+          xml.channel do
+            store_information(xml, store)
+            Spree::Product.find_each do |product|
+              next unless validate_product(product)
+
+              product.variants.each do |variant|
+                next unless validate_variant(variant, product)
+
+                xml.item do
+                  required_product_information(xml, variant, product)
```

Comment:
I would have a single method `add_product_information_to_xml`, and within that method, you can call `add_optional_product_information`, rather than calling them both from the main method.